File: vdp-java/src/test/java/com/visa/vdp/vctc/TestConsumerRules.java
package com.visa.vdp.vctc;

import org.apache.http.client.methods.CloseableHttpResponse;
import org.testng.Assert;

import java.util.HashMap;

import org.apache.http.HttpStatus;
import org.testng.annotations.BeforeTest;
import org.testng.annotations.Test;

import com.visa.vdp.utils.AbstractVisaAPIClient;
import com.visa.vdp.utils.MethodTypes;
import com.visa.vdp.utils.Property;
import com.visa.vdp.utils.VisaProperties;

public class TestConsumerRules {
	
	String cardRegisterData;
	AbstractVisaAPIClient abstractVisaAPIClient;
	
	@BeforeTest(groups = "vctc")
	public void setup() {
		this.abstractVisaAPIClient = new AbstractVisaAPIClient();
		this.cardRegisterData = 
				"{"
					 + "\"primaryAccountNumber\": \""+ VisaProperties.getProperty(Property.VCTC_TEST_PAN) +"\""
				+ "}";
	}
	
	@Test(groups = "vctc")
	public void testRegisterACard() throws Exception {
	    String baseUri = "vctc/";
	    String resourcePath = "customerrules/v1/consumertransactioncontrols";
	    
	    CloseableHttpResponse response = this.abstractVisaAPIClient.doMutualAuthRequest(baseUri + resourcePath, "Register A Card", this.cardRegisterData, MethodTypes.POST, new HashMap<String, String>());
	    Assert.assertTrue((response.getStatusLine().getStatusCode() == HttpStatus.SC_OK || response.getStatusLine().getStatusCode() == HttpStatus.SC_CREATED));
	    response.close();
	}

}

File: vdp-python/visa/test/vctc/test_consumer_rules.py
from visa.test.helpers.vdp_client_utils import VDPTestCaseClient
import json
'''
@author: visa
'''

class TestConsumerRules(VDPTestCaseClient):

    def setUp(self):
        self.card_register_data = json.loads('''{
          "primaryAccountNumber": ''' + self.config.get('VDP','vctcTestPan') +'''
        }''')
    
    def test_register_a_card(self):
        base_uri = 'vctc/'
        resource_path = 'customerrules/v1/consumertransactioncontrols'
        response = self.do_mutual_auth_request(base_uri + resource_path, self.card_register_data, 'Register a card call', 'post')
        self.assertTrue((str(response.status_code) == "200" or str(response.status_code) == "201"),"Register a card test failed")
        pass
